x-pack/plugins/security/server/authorization/validate_es_response.ts

```diff
@@ -24,35 +24,43 @@ export function validateEsPrivilegeResponse(
   return response;
 }
 
-function buildActionsValidationSchema(actions: string[]) {
-  return schema.object({
-    ...actions.reduce<Record<string, any>>((acc, action) => {
-      return {
-        ...acc,
-        [action]: schema.boolean(),
-      };
-    }, {}),
-  });
-}
-
 function buildValidationSchema(application: string, actions: string[], resources: string[]) {
-  const actionValidationSchema = buildActionsValidationSchema(actions);
+  const actionValidationSchema = schema.boolean();
+  const actionsValidationSchema = schema.object(
+    {},
+    {
+      unknowns: 'allow',
+      validate: (value) => {
+        const actualActions = Object.keys(value).sort();
+        if (
+          actions.length !== actualActions.length ||
+          ![...actions].sort().every((x, i) => x === actualActions[i])
```

Comment:
Nit: FWIW, these changes will cause this validation schema to behave differently (fail) if the consumer passes in any duplicate actions. We only have one consumer, which is already ensuring that actions are unique:

https://github.com/elastic/kibana/blob/7fc56131c0cb36371bd57386f240f53244a14e9a/x-pack/plugins/security/server/authorization/check_privileges.ts#L49

May be worth noting this in a TSdoc on the `validateEsPrivilegeResponse` function, though.